delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/routes/FilesRoutes.scala

```diff
@@ -329,15 +355,18 @@ object FilesRoutes {
       metadata: Option[FileCustomMetadata]
   )
 
+  final case class RegisterFileRequest(path: Path, metadata: Option[FileCustomMetadata])
```

Comment:
like the link operation, it would be good to leave the ability to pass the media type so that it takes precedence on the detected one.
Because there are examples of some extensions like asc which does not mean the same in the BBP context.